aws/assume-role/README.md

```diff
@@ -108,10 +108,125 @@ tasks:
 
   - key: chained-role
     use: assume-role
-    call: aws/assume-role 1.1.3
+    call: aws/assume-role 1.1.4
     with:
       source-profile-name: your-profile
       region: us-east-2
       role-to-assume: arn:aws:iam::your-account-id:role/your-other-role
       profile-name: your-other-profile
 ```
+
+## Upgrading from v1.X.X
+
+In v1.X.X the AWS OIDC token was provided as a leaf parameter.
+Starting in version 2, the AWS OIDC token is provided to tasks that use the assume role leaf task as an environment variable (default: `AWS_OIDC_TOKEN`).
+
+As a result of this, upon retrying a task, a new token will be used, preventing the incidental use of expired credentials.
+
+### Assuming a Role
+
+#### Before
+
+```yaml
+tasks:
+  - key: aws-cli
+    call: aws/install-cli 1.0.1
+
+  - key: assume-role
+    use: aws-cli
+    call: aws/assume-role 1.1.4
+    with:
+      region: us-east-2
+      role-to-assume: arn:aws:iam::your-account-id:role/your-role
+      oidc-token: ${{ vaults.your-vault.oidc.your-token }}
+
+  - key: your-task
+    use: assume-role
+    run: # ...
+```
+
+#### After
+
+```yaml
+tasks:
+  - key: aws-cli
+    call: aws/install-cli 1.0.1
+
+  - key: assume-role
+    use: aws-cli
+    call: aws/assume-role 2.0.0 # replaces aws/assume-role 1.1.4
+    with:
+      region: us-east-2
+      role-to-assume: arn:aws:iam::your-account-id:role/your-role
+
+  - key: your-task
+    use: assume-role
+    run: ...
+    env:
+      AWS_OIDC_TOKEN:
+        value: ${{ vaults.your-vault.oidc.your-token }}
+        cache-key: excluded
+```
+
+### Role Chaining
+
+#### Before
+
+```yaml
+tasks:
+  - key: aws-cli
+    call: aws/install-cli 1.0.1
+
+  - key: assume-role
+    use: aws-cli
+    call: aws/assume-role 1.1.4
+    with:
+      region: us-east-2
+      role-to-assume: arn:aws:iam::your-account-id:role/your-role
+      oidc-token: ${{ vaults.your-vault.oidc.your-token }}
+
+  - key: chain-role
+    use: assume-role
+    call: aws/assume-role 1.1.4
+    with:
+      region: us-east-2
+      role-to-assume: arn:aws:iam::your-account-id:role/your-other-role
+
+  - key: your-task
+    use: chain-role
+    run: ...
+```
+
+#### After
+
+```yaml
+tasks:
+  - key: aws-cli
+    call: aws/install-cli 1.0.1
+
+  - key: assume-role
+    use: aws-cli
+    call: aws/assume-role 2.0.0 # replaces aws/assume-role 1.1.4
+    with:
+      oidc-token: ${{ vaults.your-vault.oidc.your-token }}
+      region: us-east-2
+      role-to-assume: arn:aws:iam::your-account-id:role/your-role
+
+  - key: chain-role
+    use: aws-cli
```

Comment:
Is it worth calling out via comment this change to `use`? Does it affect anything in practice?

Reply:
https://github.com/rwx-research/mint-leaves/pull/188#discussion_r1946922123